js/util/browser/browser.js

```diff
@@ -98,7 +100,14 @@ exports.supported = function(options) {
         },
 
         function() {
-            return new Canvas().supportsWebGLContext((options && options.failIfMajorPerformanceCaveat) || false);
+            var opt = (options && options.failIfMajorPerformanceCaveat) || false,
+                fimpc = 'fimpc_' + String(opt);
+            if (exports.supportsWebGL[fimpc] === undefined) {
+                var canvas = new Canvas();
+                exports.supportsWebGL[fimpc] = canvas.supportsWebGLContext(opt);
+                canvas.canvas = undefined;
```

Comment:
Yeah it makes me feel icky too.  But I tested it in Chrome and that line made a difference.  With the memoization commented out, I'd run `for (var i=0; i<5; i++) mapboxgl.supported()` a few times (panning the map a bit and giving GC a chance to happen between each run), and it would be noticeably less likely to error, (i.e. noticeably faster at releasing that canvas holding the context).

So, that's a cool story, but i'll take the line out anyway.